pontoon/base/views.py

```diff
@@ -382,16 +382,10 @@ def approve_translation(request):
         )
 
     # Check for errors.
-    if project.slug == 'tutorial':
-        # Checks are disabled for the tutorial.
-        use_ttk_checks = False
-    else:
-        try:
-            use_ttk_checks = UserProfile.objects.get(user=user).quality_checks
-        except UserProfile.DoesNotExist:
-            use_ttk_checks = True
+    # Checks are disabled for the tutorial.
+    use_checks = project.slug != 'tutorial'
 
-    if use_ttk_checks and translation.errors.all():
+    if use_checks and translation.errors.exists():
```

Comment:
That's an improvement, but we should actually re-run checks here:

```python
failed_checks = run_checks(...)
if are_blocking_checks(failed_checks, ignore_warnings):
    return JsonResponse({
        'failedChecks': failed_checks,
    })
```

The reviewer should not only be told that translation contains errors, but also be able to see any warnings and bypass them. It should also see Translate Toolkit warnings.

Reply:
There's one thing I don't understand here. Aren't the errors and warnings saved in the database? Why do we need to re-run the checks, instead of just getting them from the database and returning that? 

I assume the reason why you need to return them is because in some cases a user will approve something that is not the active translation.

Reply:
> There's one thing I don't understand here. Aren't the errors and warnings saved in the database? Why do we need to re-run the checks, instead of just getting them from the database and returning that?

Good question. This is an edge case, but if we introduce new checks between the point when the translation was stored (along with failed checks) and the point when it gets approved, the new checks would never run. Unless we use the `run_checks` management command everytime we introduce important new checks (that's actually why we created it).

@Pike do you have thoughts on whether we should re-run checks when approving, or should we just return the ones in the DB?

> I assume the reason why you need to return them is because in some cases a user will approve something that is not the active translation.

We need to `return` them, because we need to list them in the UI. And let the user ignore them, if they are only warnings.

Reply:
We didn't have new checks in a while, outside of native Android.

Arguably, we should rerun checks when we update compare-locales, though, as stated above, that's excessive these days.

Does this bug cover un-obsoleting translations? If so, we probably want to rerun tests?

Not a strong opinion.

Reply:
Thanks for the update! And sorry for the ping, forgot you were on a PTO.

If by un-obsoleting translations you mean un-rejecting, then no. This patch only adds a new method for approving translations (currently the same method is used for saving and approving translations).

Reply:
@adngdb If Axel doesn't have a strong opinion, I'm fine with moving on here by simply returning the existing failed checks and not re-running them.